packages/core/src/test/applicationcomposer/messageHandlers/generateResourceHandler.test.ts

```diff
@@ -5,16 +5,18 @@
 
 import assert from 'assert'
 import sinon from 'sinon'
+import semver from 'semver'
 import { createTemplate, createWebviewContext } from '../utils'
 import { generateResourceHandler } from '../../../applicationcomposer/messageHandlers/generateResourceHandler'
 import { Command, MessageType } from '../../../applicationcomposer/types'
+import * as env from '../../../shared/vscode/env'
 
 describe('generateResourceHandler', function () {
-    afterEach(() => {
-        sinon.restore()
-    })
-
-    it('amazon q is not installed', async () => {
+    it('amazon q is not installed', async function () {
+        const version = env.getMinVscodeVersion()
+        if (semver.lt(version, '1.88.0')) {
+            this.skip()
+        }
```

Comment:
can use `isMinVscode('1.89.0')` https://github.com/aws/aws-toolkit-vscode/blob/d4d1ae0d065edc4ac795de32996110efb7dbf0dc/packages/core/src/shared/vscode/env.ts#L70

and that eliminates the need for the techdebt test (though that's a good instinct).

```suggestion
        if (isMinVscode('1.89.0')) {
            this.skip()
        }
```

p.s. `isMinVscode` is kinda hard to understand, maybe it should check `semver.gt` instead of `semver.gte`.

Reply:
Yeah, agree on the function. It feels like its doing two things at once: checking if we are on minimum version, and throwing an error when we exceed a threshold. Its not entirely clear to me that these tasks are related, so could be worth splitting/rewording for readability sake.